var/spack/repos/builtin/packages/fenics/package.py

```diff
@@ -100,8 +100,8 @@ class Fenics(CMakePackage):
     depends_on('boost+filesystem+program_options+system+iostreams+timer+regex+chrono@1.68.0', when='@:2018')
 
     depends_on('mpi', when='+mpi')
-    depends_on('hdf5+hl+fortran', when='+hdf5+petsc')
-    depends_on('hdf5+hl', when='+hdf5~petsc')
+    depends_on('hdf5@:1.11+hl+fortran', when='+hdf5+petsc')
+    depends_on('hdf5@:1.11+hl', when='+hdf5~petsc')
```

Comment:
Nitpick: Have you actually tried building with HDF5 1.11? As 1.11 is the development release for 1.12, I would expect it to also not build. This should therefore probably be `@:1.10`.

Reply:
No, only HDF5 1.10.7 was tried. The PR was updated.